src/contracts/GPv2AccessControl.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: LGPL-3.0-or-newer
+pragma solidity ^0.6.12;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/EnumerableSet.sol";
+
+/// @title Gnosis Protocol v2 Access Control Contract
+/// @author Gnosis Developers
+contract GPv2AccessControl is Ownable {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    EnumerableSet.AddressSet private solvers;
```

Comment:
I think solver does not need to be enumerable. storing them in a hashmap would be cheaper, right?

Reply:
I just wanted to take advantage of an existing library that implements a few methods we might want to use... like containment. Of course, we could also go with a hashmap but we would lose track of the number of solvers and also would not be able to easily access the current list of solvers.

Reply:
In a sense, this is practically equivalent to a bidirectional hashmap.

Reply:
I think having them enumerable could be very helpful to keep track who can and cannot (anymore) send solutions. Otherwise we'd have to decode all past transactions manually to retrieve this information.
In the end, the only function that needs to be gas efficient is `isSolver`, the rest can implement extra functionality.

Reply:
I will consider this settled and leave it as is.

Reply:
I also think its fine as is. This is not part of the settlement contract and I imagine this will be upgraded with DAO-specific logic later on (for example things like staking).